tools/submission/submission_checker/checks/base.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+
+class BaseCheck(ABC):
+    """
+    A generic check class meant to be inherited by concrete check implementations.
+    Subclasses must implement the `run()` method.
+    """
+
+    def __init__(self, log, path):
+        self.checks = []
+        self.log = log
+        self.path = path
+        self.name = "base checks"
+        pass
+
+    def run_checks(self):
+        """
+        Execute the check.
+        Must be implemented by subclasses.
+        Should return a CheckResult instance.
+        """
+        valid = True
+        errors = []
+        for check in self.checks:
+            v = self.execute(check)
+            valid &= v
+            if not valid:
+                return False
```

Comment:
I'm wondering if it makes more sense to run every check here and return the success value of each, keyed by the checker's class? Something like:
```
{
    AccuracyCheck: True,
    ComplianceCheck: False,
    PerformanceCheck: True,
    ...
}
```